tests/csrf_tests/tests.py

```diff
@@ -69,6 +69,79 @@ def test_mask_cipher_secret(self):
                 masked = _mask_cipher_secret(secret)
                 self.assertMaskedSecretCorrect(masked, secret)
 
+    def test_get_token_csrf_cookie_set(self):
+        """Test calling when request.META contains "CSRF_COOKIE"."""
+        request = HttpRequest()
+        request.META['CSRF_COOKIE'] = MASKED_TEST_SECRET1
+        self.assertFalse('CSRF_COOKIE_NEEDS_UPDATE' in request.META)
+        token = get_token(request)
+        self.assertNotEqual(token, MASKED_TEST_SECRET1)
+        self.assertMaskedSecretCorrect(token, TEST_SECRET)
+        # The existing cookie is preserved.
+        self.assertEqual(request.META['CSRF_COOKIE'], MASKED_TEST_SECRET1)
+        self.assertIs(request.META['CSRF_COOKIE_NEEDS_UPDATE'], True)
+
+    def test_get_token_csrf_cookie_not_set(self):
+        """Test calling when request.META does not contain "CSRF_COOKIE"."""
+        request = HttpRequest()
+        self.assertFalse('CSRF_COOKIE' in request.META)
+        self.assertFalse('CSRF_COOKIE_NEEDS_UPDATE' in request.META)
+        token = get_token(request)
+        cookie = request.META['CSRF_COOKIE']
+        self.assertEqual(len(cookie), CSRF_TOKEN_LENGTH)
+        unmasked_cookie = _unmask_cipher_token(cookie)
+        self.assertMaskedSecretCorrect(token, unmasked_cookie)
+        self.assertIs(request.META['CSRF_COOKIE_NEEDS_UPDATE'], True)
+
+    def test_rotate_token(self):
+        request = HttpRequest()
+        request.META['CSRF_COOKIE'] = MASKED_TEST_SECRET1
+        self.assertFalse('CSRF_COOKIE_NEEDS_UPDATE' in request.META)
+        rotate_token(request)
+        # Check that the underlying secret was changed.
+        cookie = request.META['CSRF_COOKIE']
+        self.assertEqual(len(cookie), CSRF_TOKEN_LENGTH)
+        unmasked_cookie = _unmask_cipher_token(cookie)
+        self.assertNotEqual(unmasked_cookie, TEST_SECRET)
+        self.assertIs(request.META['CSRF_COOKIE_NEEDS_UPDATE'], True)
+
+    def test_sanitize_token_masked(self):
+        """Test passing a token of length CSRF_TOKEN_LENGTH."""
+        cases = [
+            (MASKED_TEST_SECRET1, MASKED_TEST_SECRET1),
+            (64 * 'a', 64 * 'a'),
+        ]
+        for token, expected in cases:
+            with self.subTest(token=token):
+                actual = _sanitize_token(token)
+                self.assertEqual(actual, expected)
+
+    def test_sanitize_token_unmasked(self):
+        """Test passing a token of length CSRF_SECRET_LENGTH."""
```

Comment:
`Test passing` is unnecessary, IMO, maybe:
```suggestion
        # A token of length CSRF_SECRET_LENGTH.
```